lib/oban.ex

```diff
@@ -745,15 +745,33 @@ defmodule Oban do
 
       Oban.drain_queue(:default)
       %{success: 2, failure: 1}
+
+  ## Scheduled Jobs
+
+  By default, `drain_queue/3` will execute all currently available jobs. In order to execute
+  scheduled jobs, you may pass the `:with_scheduled` flag which will cause scheduled jobs
+  to be marked as `available` beforehand.
+
+      # This will execute all scheduled jobs.
+      Oban.drain_queue(:default, with_scheduled: true)
+      %{success: 1, failure: 0}
+
   """
   @doc since: "0.4.0"
-  @spec drain_queue(name :: atom(), queue :: atom() | binary()) :: %{
+  @spec drain_queue(name :: atom(), queue :: atom() | binary(), opts :: keyword()) :: %{
           success: non_neg_integer(),
           failure: non_neg_integer()
         }
-  def drain_queue(name \\ __MODULE__, queue)
-      when is_atom(name) and (is_atom(queue) or is_binary(queue)) do
-    Producer.drain(to_string(queue), config(name))
+  def drain_queue(name \\ __MODULE__, queue, opts \\ [])
```

Comment:
This can be done seamlessly with manual clauses:

```elixir
  def drain_queue(queue) when is_atom(queue) or is_binary(queue) do
    drain_queue(__MODULE__, queue, [])
  end

  def drain_queue(name, queue) when is_atom(name) and (is_atom(queue) or is_binary(queue)) do
    drain_queue(name, queue, [])
  end

  def drain_queue(queue, opts) when (is_atom(queue) or is_binary(queue)) and is_list(opts) do
    drain_queue(__MODULE__, queue, opts)
  end

  def drain_queue(name, queue, opts)
      when is_atom(name) and (is_atom(queue) or is_binary(queue)) and is_list(opts) do
    Producer.drain(to_string(queue), config(name))
  end
```

It may also require some additional `drain_queue` specs.

Reply:
Yes, this is more or less what I meant to say in 2. Reads nice, but don't you find it a bit irritating as an interface? Specifically that there is a `drain_queue/2` that has the `queue` as the first or as the second argument. If I were to `@spec` and document this function, the console help text would look like

```elixir
iex(12)> h Oban.drain_queue/2

                          def drain_queue(name, queue)

  @spec drain_queue(atom() | binary(), atom() | binary() | list()) :: map()

Drain a queue. See drain_queue/3.
```

The headline shows the first implementation only, and the signature looks interesting, but very non-descriptive (i.e. it doesn't convey which combinations of types work).

Reply:
Ah, now that I've re-read your second option I see that is what you were saying. I don't find it the most appealing interface, and it is a little bit fragile.

While we are technically pre 1.0, I'd prefer not to break things too drastically. Option three is interesting in general (and exactly what I did with Kiq). That seems like overkill for what we're trying to achieve here.

What about another option that isn't so ugly and is only slightly breaking—rename `drain_queue` as `drain`. That would change the `queue` argument to `opts` and enforce `queue` as a required argument.

```elixir
Oban.drain(queue: "default", with_scheduled: true)
```

Reply:
This sounds nice - would you want to move the documentation though?

Reply:
Pushed changes with a midway solution for documentation.